test/controllers/presroc/bill_runs.controller.test.js

```diff
@@ -143,4 +143,46 @@ describe('Presroc Bill Runs controller', () => {
       })
     })
   })
+
+  describe('Generate a bill run summary: POST /v2/{regimeId}/bill-runs/{billRunId}/generate-summary', () => {
+    const options = (token, payload, billRunId) => {
+      return {
+        method: 'POST',
+        url: `/v2/wrls/bill-runs/${billRunId}/generate-summary`,
+        headers: { authorization: `Bearer ${token}` },
+        payload: payload
+      }
+    }
+
+    beforeEach(async () => {
+      billRun = await BillRunHelper.addBillRun(authorisedSystem.id, regime.id)
+    })
+
+    describe('When the request is valid', () => {
+      describe('When the summary has not yet been generated', () => {
+        it('returns success status 204', async () => {
+          const requestPayload = GeneralHelper.cloneObject(requestFixtures.simple)
+
+          const response = await server.inject(options(authToken, requestPayload, billRun.id))
+
+          expect(response.statusCode).to.equal(204)
+        })
+      })
+    })
+
+    describe('When the request is invalid', () => {
+      describe('When the summary has already been generated', () => {
+        it('returns error status 409', async () => {
+          const requestPayload = GeneralHelper.cloneObject(requestFixtures.simple)
+          await server.inject(options(authToken, requestPayload, billRun.id))
+
+          const response = await server.inject(options(authToken, requestPayload, billRun.id))
```

Comment:
Just to note that the test currently works because we don't actually change the status after setting it to `generating`! Good shout on how we ought to be testing it though, I'll go ahead and make the change.